### PR TITLE
Java servlet for sending user notifications and matches to Javascript for rendering.

### PR DESCRIPTION
**Pull request by lailagamaleldin**
Updated to render comments and notifications to page.

### CODE REVIEW

File: step/src/main/java/com/google/sps/servlets/HomepageServlet.java [hunk 1]
```diff
@@ -1,47 +1,71 @@
...
+    }
 
-//   }
+    int notificationCounter = 0;
+    for (Notification notification: notifications) {
+      if (notificationCounter == NUM_NOTIFS_TO_DISPLAY) {
+        break;
+      }
+      notificationsArray.add(notification.getText());
+    }
```
Comment (adamsmlsn) [inline step/src/main/java/com/google/sps/servlets/HomepageServlet.java]: Doesn't look like notificationCounter is being incremented.

File: step/src/main/webapp/profile.jsp [hunk 2]
```diff
@@ -26,17 +26,26 @@ limitations under the License.
...
         response.sendRedirect("index.jsp");   
     } %>
+
+    <script>
+      function getMatches() {
+        fetch('/Homepage')
+          .then((response) => {
+            // TODO: print to screen
+            console.log(response.json());
+          });
```
Comment (adamsmlsn) [inline step/src/main/webapp/profile.jsp]: How are the user's matches being displayed on the front end?

**Review by adamsmlsn — CHANGES_REQUESTED** [review]
There's some issues with this PR. When I try running it, I get database errors that aren't finding the user (the same problem we had earlier today). Can you merge master into this branch so this branch will have the database fix? Then I'll be able to check how the changes look on the front end.

File: step/src/main/webapp/profile.jsp [hunk 3]
```diff
@@ -42,10 +53,58 @@ limitations under the License.
...
-            // TODO: print to screen
-            console.log(response.json());
+            return response.json();
+          })
+          .then((json) => {
+            matches = json["matches"];
+            notifications = json["notifications"];
+
+            renderMatches(matches);
+            renderNotifications(notifications);
```
Comment (adamsmlsn) [inline step/src/main/webapp/profile.jsp:146]: Good structuring!

**Review by adamsmlsn — APPROVED** [review]
LGTM. When you merge, make sure the changes from [this datastore commit](https://github.com/googleinterns/step21-2020/commit/c7eeb1a18fc6aa6f11583d50ea71221499f7490f) are persistent. I think git might undo it when you merge since it's not in your branch right now.

**Review by nganttran — APPROVED** [review]

**Review by wangding-google — APPROVED** [review]

**Pull request merged**